string name
bool first_info
---
bool res
